CesiumGltfWriter/src/AnimationWriter.cpp

```diff
@@ -0,0 +1,104 @@
+#include "AnimationWriter.h"
+#include "ExtensionWriter.h"
+#include "JsonObjectWriter.h"
+#include <CesiumGltf/AnimationChannel.h>
+#include <CesiumGltf/AnimationChannelTarget.h>
+#include <magic_enum.hpp>
+#include <stdexcept>
+#include <type_traits>
+
+void writeAnimationChannel(
+    const CesiumGltf::AnimationChannel& animationChannel,
+    CesiumGltf::JsonWriter& jsonWriter) {
+  auto& j = jsonWriter;
+  j.StartObject();
+  j.Key("sampler");
+  j.Int(animationChannel.sampler);
+  j.Key("target");
+  {
+    j.StartObject();
+    j.Key("node");
+    j.Int(animationChannel.target.node);
+    j.Key("path");
+    const auto pathAsString =
+        std::string(magic_enum::enum_name(animationChannel.target.path));
+    j.String(pathAsString.c_str());
+
+    if (!animationChannel.target.extensions.empty()) {
+      writeExtensions(animationChannel.target.extensions, j);
+    }
+
+    if (!animationChannel.target.extras.empty()) {
+      j.Key("extras");
+      CesiumGltf::writeJsonValue(animationChannel.target.extras, j, false);
+    }
+  }
+  j.EndObject();
+}
+
+void writeAnimationSampler(
+    const CesiumGltf::AnimationSampler& animationSampler,
+    CesiumGltf::JsonWriter& jsonWriter) {
+  auto& j = jsonWriter;
+
+  j.StartObject();
+  j.KeyPrimitive("input", animationSampler.input);
+  j.KeyPrimitive(
+      "interpolation",
+      magic_enum::enum_name(animationSampler.interpolation));
+  j.KeyPrimitive("output", animationSampler.output);
+
+  if (!animationSampler.extensions.empty()) {
+    writeExtensions(animationSampler.extensions, j);
+  }
+
+  if (!animationSampler.extras.empty()) {
+    j.Key("extras");
+    CesiumGltf::writeJsonValue(animationSampler.extras, j, false);
+  }
+
+  j.EndObject();
+}
+
+void CesiumGltf::writeAnimation(
+    const std::vector<Animation>& animations,
+    CesiumGltf::JsonWriter& jsonWriter) {
+
+  if (animations.empty()) {
+    return;
+  }
+
+  auto& j = jsonWriter;
+  j.Key("animations");
+  j.StartArray();
+  for (const auto& animation : animations) {
+    j.StartObject();
+    if (animation.channels.empty()) {
+      throw std::runtime_error("Must have at least 1 animation channel");
```

Comment:
We've generally worked on the assumption that cesium-native will be used in some environments where exceptions are disabled. And so exceptions shouldn't be the primary means of reporting errors. Especially in situations like this one where it might indicate user error more than a truly exceptional circumstance.

IMO, something like this should be reported as a warning, and say "The generated glTF will not be spec-compliant" rather than refusing to write a glTF at all.

Reply:
Using `std::cerr` now. Clients might get annoyed having random stuff spewed on `stderror` though, let me know if you have a specific implementation in mind for error handling in an exceptionless land. (Although your only real options are logging stuff and returning error codes, so I guess I'm asking which is preferred).

Reply:
CesiumGltfReader returns a list of errors and warnings as part of its result. I imagine the same should work well here.

Reply:
updated API to mimic CesiumGltfReader's `ModelReaderResult`